src/modules/companies/companies.service.ts

```diff
@@ -70,13 +71,33 @@ export class CompaniesService {
       const company = await this.findOne(id);
 
       if (!company)
-        throw new NotFoundException(`Company with ID ${id} not found`);
-
+        throw new NotFoundException(`Company with ID ${id} is not found`);
+
+      if (
+        updateCompaniesDto.organization_name &&
+        (await this.companyRepository.findOne({
+          where: { organization_name: updateCompaniesDto.organization_name },
+        }))
+      ) {
+        throw new BadRequestException('Such company already exists');
+      }
+
+      if (
+        updateCompaniesDto.email &&
+        (await this.companyRepository.findOne({
+          where: { email: updateCompaniesDto.email },
+        }))
```

Comment:
same

Reply:
I suggest get all the data in one query and use results in different conditions if it possible